src/core/crypto_tm.c

```diff
@@ -212,10 +234,10 @@ int32_t Crypto_TM_ApplySecurity(SecurityAssociation_t* sa_ptr)
         printf("\n" RESET);
     }
 #endif
-
-    if(sa_service_type != SA_PLAINTEXT && sa_ptr->ecs_len == CRYPTO_CIPHER_NONE && sa_ptr->acs_len == CRYPTO_MAC_NONE)
+    if(sa_service_type != SA_PLAINTEXT && sa_ptr->ecs_len == 0 && sa_ptr->acs_len ==0)
     {
         status = CRYPTO_LIB_ERR_NULL_CIPHERS;
+        printf("CRYPTO_LIB_ERR_NULL_CIPHERS, Invalid cipher lengths, %d\n", CRYPTO_LIB_ERR_NULL_CIPHERS);
```

Comment:
May want to wrap in #ifdef debug. Additionally, `mc_log` should handle printing the error code but extra printouts are nice!